bin/sozo/src/commands/options/account/controller.rs

```diff
@@ -237,3 +246,61 @@ fn get_dojo_world_address(
         }
     }
 }
+
+/// This function will call the `cartridge_deployController` method to deploy the account if it
+/// doesn't yet exist on the chain. But this JSON-RPC method is only available on Katana deployed on
+/// Slot. If the `rpc_url` is not a Slot url, it will return an error.
+///
+/// `cartridge_deployController` is not a method that Katana itself exposes. It's from a middleware
+/// layer that is deployed on top of the Katana deployment on Slot. This method will deploy the
+/// Contract contract of a user based on the Slot deployment.
+async fn deploy_account_if_not_exist(
+    rpc_url: Url,
+    provider: &impl Provider,
+    chain_id: Felt,
+    address: Felt,
+    username: &str,
+) -> Result<()> {
+    use reqwest::Client;
+    use serde_json::json;
+
+    // Check if the account exists on the chain
+    match provider.get_class_at(BlockId::Tag(BlockTag::Pending), address).await {
+        Ok(_) => Ok(()),
+
+        // if account doesn't exist, deploy it by calling `cartridge_deployController` method
+        Err(err @ StarknetError(ContractNotFound)) => {
+            trace!(
+                %username,
+                chain = format!("{chain_id:#}"),
+                address = format!("{address:#x}"),
+                "Controller does not exist on chain. Attempting to deploy..."
+            );
+
+            // Skip deployment if the rpc_url is not a Slot instance
+            if !rpc_url.host_str().map_or(false, |host| host.contains("api.cartridge.gg")) {
+                warn!(%rpc_url, "Unable to deploy Controller on non-Slot instance.");
+                bail!("Controller with username '{username}' does not exist: {err}");
+            }
+
+            let response = Client::new()
+                .post(rpc_url)
+                .json(&json!({
+                    "id": 1,
+                    "jsonrpc": "2.0",
+                    "method": "cartridge_deployController",
+                    "params": { "id": username },
+                }))
+                .send()
+                .await?;
+
+            if !response.status().is_success() {
+                bail!("Failed to deploy controller: {}", response.status());
```

Comment:
Is the status enough to know what failed? Or should we also output the `data` from json RPC error type?

Reply:
yeah make sense. we could call `.json()` on the response, which would return error if its not a JSON but should be fine as JSON-RPC is expected to be in JSON format.